### DISCUSSION TITLE
[Bug] java.lang.UnsupportedOperationException: class redefinition failed: attempted to change superclass or interfaces

### DISCUSSION THREAD

**Discussion by muwn** [apache/skywalking#11751] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

spring boot   2.3.12.RELEASE
spring cloud Hoxton.SR1

After the project adds the eurkea registration center dependency spring-cloud-starter-netflix-eureka-client, the problem appears at startup, as shown in the picture below

<img width="1478" alt="image" src="https://github.com/apache/skywalking/assets/9361471/25f292eb-3971-41b0-9c2d-afecd688e9c9">

### What you expected to happen

Normal start

### How to reproduce

 spring boot  2.3.12.RELEASE
spring cloud Hoxton.SR12
idea 2023.3.2 Build #IU-233.13135.103, built on December 20, 2023

<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-starter-netflix-eureka-client</artifactId>
</dependency>

JVM options
-javaagent:/Users/xxxx/Downloads/skywalking-agent/skywalking-agent.jar

environment variables 
SW_AGENT_COLLECTOR_BACKEND_SERVICES=192.168.2.xxx:11800;SW_AGENT_NAME=java-example-local

### Are you willing to submit a pull request to fix on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Do you add spring devtool or something in your classpath? Retransfer means you have some other agents running and manipulating source codes.

**Reply by muwn** [reply]
pom.xml

```
<dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>cn.hutool</groupId>
            <artifactId>hutool-all</artifactId>
            <version>5.7.19</version>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.22</version>
        </dependency>
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.22</version>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-netflix-eureka-client</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.apache.skywalking</groupId>
            <artifactId>apm-toolkit-trace</artifactId>
            <version>8.16.0</version>
        </dependency>
```

**Reply by wu-sheng** [reply]
I can't verify lib for you, but that is what your error means, from my guessing.

**Reply by muwn** [reply]
<img width="1010" alt="image" src="https://github.com/apache/skywalking/assets/9361471/90614267-a2bb-4665-bf41-cf1e30cadc7a">

**Comment by muwn** [comment]
```

        <dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-starter-netflix-eureka-client</artifactId>
            <exclusions>
                <exclusion>
                    <artifactId>reactor-core</artifactId>
                    <groupId>io.projectreactor</groupId>
                </exclusion>
            </exclusions>
        </dependency>
```
I tried this method and it works normally, but I don't know if it will affect other businesses.
